lib/wallet/send.dart

```diff
@@ -53,30 +53,28 @@ class _SendState extends State<Send> {
                       children: [
                         Row(
                           mainAxisAlignment: MainAxisAlignment.end,
-                          children: [
-                            ElevatedButton(
-                                onPressed: () async {
-                                  FLog.info(text: "Sending invoice" + invoice);
-                                  String? error;
-                                  try {
-                                    await api.sendLightningPayment(invoice: invoice);
-                                  } on FfiException catch (e) {
-                                    error = e.message;
-                                  }
-
-                                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
-                                    content: Text(error == null
-                                        ? "Paid lightning invoice $invoice"
-                                        : "Failed to send payment: $error"),
-                                  ));
-                                  context.go('/');
-                                },
-                                child: const Text('Send'))
-                          ],
+                          children: [SubmitButton(onPressed: send, label: "Pay Invoice")],
                         ),
                       ],
                     ),
                   ),
                 ]))));
   }
+
+  Future<void> send() async {
+    FLog.info(text: "Paying invoice $invoice");
+    await api.sendLightningPayment(invoice: invoice).then((value) async {
```

Comment:
❓ Does the `then` callback need to be async?

Reply:
Since we are not doing `async` operations in this `then` block we don't need it, nice spotting :)
I removed it for `send_on_chain` as well :)